test/graphql/execution/executor_test.exs

```diff
@@ -37,6 +37,21 @@ defmodule GraphQL.Execution.Executor.ExecutorTest do
     assert_execute {~S[{ greeting(name: "Elixir") }], TestSchema.schema}, %{greeting: "Hello, Elixir!"}
   end
 
+  test "anonymous fragments are processed" do
+    schema = %GraphQL.Schema{
+      query: %GraphQL.ObjectType{
+        name: "X",
+        fields: %{
+          id:   %{type: "String", resolve: fn(p, _, _) -> p.id   end},
```

Comment:
Default resolve should just work in this case, so you can drop the resolves completely